syntax = "proto3";
package pb;
option go_package = "internal/pb";

service UserService {
    rpc CreateUser (CreateUserRequest) returns (CreateUserResponse);
    rpc UpdateUser (UpdateUserRequest) returns (UpdateUserResponse);
    rpc DeleteUser (DeleteUserRequest) returns (DeleteUserResponse);
}

message User {
    int32 id = 1;
    string name = 2;
    string email = 3;
    string password = 4;
}

message CreateUserRequest {
    User user = 1;
}

message CreateUserResponse {
    User user = 1;
}

message UpdateUserRequest {
    User user = 1;
}

message UpdateUserResponse {
    User user = 1;
}

message DeleteUserRequest {
    int32 id = 1;
}

message DeleteUserResponse {
    bool success = 1;
}
